Modules/Scripted/DICOMLib/DICOMPlugin.py

```diff
@@ -140,19 +156,48 @@ def export(self,exportable):
     """
     return ""
 
-  def defaultSeriesNodeName(self,seriesUID):
+  def getVolumeName(self, files):
     """Generate a name suitable for use as a mrml node name based
-    on the series level data in the database"""
-    instanceFilePaths = slicer.dicomDatabase.filesForSeries(seriesUID)
-    if len(instanceFilePaths) == 0:
-      return "Unnamed Series"
-    seriesDescription = slicer.dicomDatabase.fileValue(instanceFilePaths[0],self.tags['seriesDescription'])
-    seriesNumber = slicer.dicomDatabase.fileValue(instanceFilePaths[0],self.tags['seriesNumber'])
-    name = seriesDescription
-    if seriesDescription == "":
-      name = "Unnamed Series"
-    if seriesNumber != "":
-      name = seriesNumber + ": " + name
+    on the series level data in the database.
+
+    Arguments
+
+      files: files that will be used to build the volume name using the DICOM tags in a list of DICOM files
+    """
+    if not files or len(files) == 0:
+      return "UNNAMED VOLUME"
```

Comment:
Yap. The idea is to stress the cases in which the volume naming is not correct for any reason